hs_communities/management/commands/check_czo_groups.py

```diff
@@ -0,0 +1,208 @@
+"""
+Check that CZO groups are set up properly.
+If a resource is owned by a CZO owner, and not part of the CZO group,
+  then add it to the group.
+If a resource is not owned by CZO national, make it owned by CZO national.
+"""
+
+from django.core.management.base import BaseCommand
+from django.contrib.auth.models import User, Group
+from hs_access_control.models import UserResourcePrivilege, GroupResourcePrivilege, \
+        GroupCommunityPrivilege, PrivilegeCodes, Community
+from hs_core.models import BaseResource
+
+
+czo_setup = [
+    ["czo_national", "CZO National", "Cross-CZO"],
+    ["czo_boulder", "CZO Boulder", "BCCZO"],
+    ["czo_calhoun", "CZO Calhoun", "CCZO"],
+    ["czo_catalina-jemez", "CZO Catalina-Jemez", "CJCZO"],
+    ["czo_eel", "CZO Eel", "ERCZO"],
+    ["czo_luquillo", "CZO Luquillo", "LCZO"],
+    ["czo_reynolds", "CZO Reynolds", "RCCZO"],
+    ["czo_shale-hills", "CZO Shale-Hills", "SSHCZO"],
+    ["czo_sierra", "CZO Sierra", "SSCZO"],
+    ["czo_christina", "CZO Christina", "CRBCZO"],
+]
+
+
+class Command(BaseCommand):
+    help = "check czo setup"
+
+    def handle(self, *args, **options):
+        national = User.objects.get(username='czo_national')
+        prefixes = {}
+        community = Community.objects.get(name='CZO National Community')
+
+        for czo in czo_setup:  # index by prefix
+            prefixes[czo[2]] = czo
+
+        for czo in czo_setup:
+            username = czo[0]
+            groupname = czo[1]
+            prefix = czo[2]  # prefix for all titles for this group.
+
+            print("checking user {} against group {}".format(username, groupname))
+            user = User.objects.get(username=username)
+            group = Group.objects.get(name=groupname)
+
+            # first check that group is in the community
+            if not Community.objects.filter(c2gcp__community=community, c2gcp__group=group).exists():
+                print("group {} is not in community {}".format(group.name, community.name))
+                # fix it NOW
+                GroupCommunityPrivilege.share(group=group, community=community,
+                                              privilege=PrivilegeCodes.VIEW,
+                                              grantor=national)
+
+            user_resources = set(BaseResource.objects.filter(r2urp__user=user))
+            group_resources = set(BaseResource.objects.filter(r2grp__group=group))
+            print("  There are {} user resources".format(len(user_resources)))
+            print("  There are {} group resources".format(len(group_resources)))
+
+            if username != 'czo_national':
+                # For every group but the master group, user should own all group resources
```

Comment:
This is the opposite of what is happening.  Line 70 makes all user resources owned by the group.

Reply:
Fixed comment.